kubernetes/skaffold/src/main/kotlin/com/google/kubernetes/tools/skaffold/SkaffoldExecutorService.kt

```diff
@@ -47,19 +47,20 @@ abstract class SkaffoldExecutorService {
     /** Path for Skaffold executable, any form supported by [ProcessBuilder] */
     protected abstract var skaffoldExecutablePath: Path
 
-    fun getSystemPath(): String = System.getenv("PATH")
-
+    /**
+     * Checks if Skaffold is available by executing a 'skaffold version' command. If the process
+     * exit code is 0, then Skaffold is considered available.
+     */
     fun isSkaffoldAvailable(): Boolean {
-        val skaffoldPathOverride = KubernetesSettingsService.instance.skaffoldExecutablePath
+        return try {
+            val process = executeSkaffold(SkaffoldExecutorSettings(ExecutionMode.VERSION)).process
+            process.waitFor()
```

Comment:
note this executes on the order of 10ms so we should be ok

Reply:
you're right but it might be good to use these params just to be safe (and i guess for other people reading it to know it wont go on for more than 10 secs):

@param timeout -  the maximum time to wait
@param unit - the time unit of the {@code timeout} argument

Reply:
great points. Let me refactor a bit to make it more robust

Reply:
updated. also with additional unit test